src/components/atoms/Typography.tsx

```diff
@@ -9,59 +10,38 @@ interface IText {
   [property: string]: FC<IBold>;
 }
 
+interface ITypographyBase extends IBold {
+  variant: Variant;
+  component: ElementType;
+}
```

Comment:
You can improve it by including `bold` in `ITypographyBase` and removing `IBold`